pandas/core/indexes/base.py

```diff
@@ -800,24 +800,29 @@ def __deepcopy__(self, memo=None):
             memo = {}
         return self.copy(deep=True)
 
-    def _validate_names(self, name=None, names=None, deep=False):
+    def _handle_names(self, name=None, names=None, deep=False):
         """
         Handles the quirks of having a singular 'name' parameter for general
         Index and plural 'names' parameter for MultiIndex.
         """
         from copy import deepcopy
-        if names is not None and name is not None:
-            raise TypeError("Can only provide one of `names` and `name`")
-        elif names is None and name is None:
+
+        if names is None and name is None:
             return deepcopy(self.names) if deep else self.names
-        elif names is not None:
-            if not is_list_like(names):
-                raise TypeError("Must pass list-like as `names`.")
-            return names
-        else:
-            if not is_list_like(name):
-                return [name]
-            return name
+
+        return self._validate_names(name=name, names=names)
+
```

Comment:
maybe if you these have doc-strings I would be able to tell what is going on here.